geth/jail/jail_cell_test.go

```diff
@@ -1,93 +1,12 @@
 package jail_test
 
 import (
-	"net/http"
-	"net/http/httptest"
 	"time"
 
 	"github.com/robertkrimen/otto"
-	"github.com/status-im/status-go/geth/jail"
 	"github.com/status-im/status-go/geth/params"
 )
 
-func (s *JailTestSuite) TestJailTimeoutFailure() {
-	require := s.Require()
-	require.NotNil(s.jail)
-
-	newCell, err := s.jail.NewJailCell(testChatID)
-	require.NoError(err)
-	require.NotNil(newCell)
-
-	// Attempt to run a timeout string against a JailCell.
-	_, err = newCell.RunOnLoop(`
-		setTimeout(function(n){
-			if(Date.now() - n < 50){
-				throw new Error("Timedout early");
-			}
-
-			return n;
-		}, 30, Date.now());
-	`)
-
-	require.NotNil(err)
-}
-
-func (s *JailTestSuite) TestJailTimeout() {
-	require := s.Require()
-	require.NotNil(s.jail)
-
-	newCell, err := s.jail.NewJailCell(testChatID)
-	require.NoError(err)
-	require.NotNil(newCell)
-
-	// Attempt to run a timeout string against a JailCell.
-	res, err := newCell.RunOnLoop(`
-		setTimeout(function(n){
-			if(Date.now() - n < 50){
-				throw new Error("Timedout early");
-			}
-
-			return n;
-		}, 50, Date.now());
-	`)
-
-	require.NoError(err)
-	require.NotNil(res)
-}
```

Comment:
I'd leave this. We don't have any other evidence that it works, actually.

Reply:
I'd love to, but these tests uses methods `RunOnLoop()` and `CallOnLoop()` which aren't used anywhere else and are incompatible with the single loop approach.

Reply:
So let's change to just `Run` and `Call`? We need these tests anyway, we want to be sure that `setTimeout` and `setInterval` works properly on our vm.

Reply:
@tiabc this test doesn't make sense with `Run` and `Call`, as they return immediately (callback function is executed in the loop).

Reply:
Let's rewrite it with delays so that it makes sense?

Reply:
@tiabc do you think it's reliable enough? Usually tests with introduced delays ends up with false negatives/positives, especially when running in heavy-loaded or restricted hosts/containers/vms.

Reply:
No, I don't think they're reliable but we can adjust delays to decrease probability of false positives/negatives. For example, `setTimeout` for 50ms and check after 1000ms that it executed once. The same about `setInterval` that it executed more than once.

I also doubt such unreliable solutions but how else can we ensure it's working? If we see any constant problems, we'll revert it, of course.

Reply:
Okey, convincing enough. Let's try this way.